Fix RHEL build (#148)
triton-inference-server/pytorch_backend: CMakeLists.txt

  endif()

    # Look for installed Torchvision package in lib paths
  find_library( LIBTORCHVISION libtorchvision.so libtorchvision.so.1 PATHS ${TRITON_PYTORCH_LIB_PATHS} )
  if(NOT ${LIBTORCHVISION}) 
    message(WARNING "TRITON_PYTORCH_ENABLE_TORCHVISION is on, but TRITON_PYTORCH_LIB_PATHS does not contain Torchvision package")
  endif(NOT ${LIBTORCHVISION})

>   endif()

Reviewer: Is this `endif()`  needs to be here?

Author: Removed it.

  endif()

    # Look for installed Torchvision package in lib paths

>   if(TRITON_PYTORCH_ENABLE_TORCHVISION)
>     if(${RHEL_BUILD})
>       if(NOT EXISTS "${TRITON_PYTORCH_LIB_PATHS}/libtorchvision.so")
>         message(WARNING "TRITON_PYTORCH_ENABLE_TORCHVISION is on, but TRITON_PYTORCH_LIB_PATHS does not contain Torchvision package")
>       endif()
>     else()
>       if(NOT EXISTS "${TRITON_PYTORCH_LIB_PATHS}/libtorchvision.so.1")
>         message(WARNING "TRITON_PYTORCH_ENABLE_TORCHVISION is on, but TRITON_PYTORCH_LIB_PATHS does not contain Torchvision package")
>       endif()
>     endif()

Reviewer: Original statement wasn't written right. Expanding that statement in such a way may be not the best way. 
Instead it should be:
```suggestion
  find_library( LIBTORCHVISION libtorchvision.so libtorchvision.so.1 PATHS ${TRITON_PYTORCH_LIB_PATHS} )
  if(NOT ${LIBTORCHVISION}) 
    message(WARNING "TRITON_PYTORCH_ENABLE_TORCHVISION is on, but TRITON_PYTORCH_LIB_PATHS does not contain Torchvision package")
  endif(NOT ${LIBTORCHVISION})
```

)

if (${TRITON_PYTORCH_ENABLE_TORCHVISION})

>   if(${RHEL_BUILD})
>     set(PT_LIBS
>         ${PT_LIBS}
>         "libtorchvision.so"
>     )
>   else()
>     set(PT_LIBS
>         ${PT_LIBS}
>         "libtorchvision.so.1"
>     )
>   endif()

Reviewer: Using [generator-expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) is more preferable in my opinion: 
```suggestion
  set(PT_LIBS
         ${PT_LIBS}
         $<IF:$<BOOL:RHEL_BUILD>,libtorchvision.so,libtorchvision.so.1>
  )
```
see: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#conditional-expressions

    "libopencv_calib3d.so"
    "libopencv_flann.so"
    "libopencv_features2d.so"
)

> if(${RHEL_BUILD})
>     set(OPENCV_LIBS
>         ${OPENCV_LIBS}
>         "libjpeg.so.62"  # RHEL uses only .so.62 version
>         "libpng16.so.16" # RHEL uses .so.16 version
>     )
> else()
>     set(OPENCV_LIBS
>         ${OPENCV_LIBS}
>         "libjpeg.so"
>         "libpng16.so"
>     )
> endif()

Reviewer: Using [generator-expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) is more preferable in my opinion: 
```suggestion
  set(OPENCV_LIBS
         ${OPENCV_LIBS}
         $<IF:$<BOOL:RHEL_BUILD>,libjpeg.so.62,libjpeg.so>
         $<IF:$<BOOL:RHEL_BUILD>,libpng16.so.16,libpng16.so>
  )
```
see: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#conditional-expressions

Author: From my perspective, using if/else is more readable and evaluates during the configuration phase. However, I just noticed that we're adopting [generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) across other backends outside of PyTorch. If we want to maintain uniformity in our approach, I can make the changes.

Reviewer: Feel free to ignore my suggestions and resolve conversations as needed.

    "libopencv_calib3d.so"
    "libopencv_flann.so"
    "libopencv_features2d.so"
)
if(${RHEL_BUILD})
    set(OPENCV_LIBS
        ${OPENCV_LIBS}
        "libjpeg.so.62"  # RHEL uses only .so.62 version
        "libpng16.so.16" # RHEL uses .so.16 version
    )
else()
    set(OPENCV_LIBS
        ${OPENCV_LIBS}
        "libjpeg.so"     # Non-RHEL uses both .so and .so.62

>         "libjpeg.so.62"

Reviewer: Why two libraries? I'm afraid it can be same file (resolved by symbolic links) isn't?
If it's different files then question arise because these two lib may have same methods within and proper precedence may require.

Author: Thanks @mc-nv for catching this. For non RHEL build it should only be libjpeg.so.

  set(TRITON_PYTORCH_LIBS "${CMAKE_CURRENT_BINARY_DIR}/libtorch.so")

  if (${TRITON_PYTORCH_ENABLE_TORCHVISION})

>     if(${RHEL_BUILD})
>       set(TRITON_PYTORCH_LIBS
>           ${TRITON_PYTORCH_LIBS}
>           "${CMAKE_CURRENT_BINARY_DIR}/libtorchvision.so")
>     else()
>       set(TRITON_PYTORCH_LIBS
>           ${TRITON_PYTORCH_LIBS}
>           "${CMAKE_CURRENT_BINARY_DIR}/libtorchvision.so.1")
>     endif()

Reviewer: Using [generator-expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) is more preferable in my opinion: 
```suggestion
  set(TRITON_PYTORCH_LIBS
          ${TRITON_PYTORCH_LIBS}
     $<IF:$<BOOL:RHEL_BUILD>,${CMAKE_CURRENT_BINARY_DIR}/libtorchvision.so,${CMAKE_CURRENT_BINARY_DIR}/libtorchvision.so.1>)
  )
```
see: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#conditional-expressionsI
